WordPress/wordcamp.org
# Speaker Feedback: Register API endpoints for feedback comments

Extends the `WP_REST_Comments_Controller` class to create a new custom endpoint for creating feedback comments.

We can't use the standard Comments endpoint, because it won't allow creation of comments with a custom type or without anything in the content field.

This also introduces schema and functions for the comment meta values that we'll be using to store feedback content, as well as sets up this plugin for phpunit testing.

Fixes #337, Refs #340 

**To test:**

1. Run all the phpunit tests and make sure all are passing.
1. It's easiest to use a tool like [Insomnia](https://insomnia.rest/) to craft payloads to send to the endpoint. The endpoint URL will be `[your-test-site].wordcamp.org/wp-json/wordcamp-speaker-feedback/v1/feedback`
1. The parameters required for a successful payload are:
    * `post` (the ID of the post to attach the feedback comment to)
    * `author` (the numeric ID of a user)
    * `meta[rating]` (an integer representation of 1 to 5 stars)
    * `meta[q1]` (a string less than 5000 characters)
1. You can use `author_name` and `author_email` (string values) in lieu of `author`.
1. You can also include `meta[q2]` and `meta[q3]` parameters, but these are not currently required fields.
1. The response to a successful request to the endpoint should be a `201` status. For an unsuccessful request, it'll be a JSON object with an error code and message. Try to get some of each! Try multiple invalid fields in one request!
1. It would probably also be good to check that the comments and their meta are actually being created in the database upon successful request.
